src/typescript/frontend/src/components/trade/DepositWithdrawModal/SelectMarketContent.tsx

```diff
@@ -7,109 +7,210 @@ import {
   getFilteredRowModel,
   useReactTable,
 } from "@tanstack/react-table";
+import { useRouter } from "next/router";
 import { useMemo, useState } from "react";
 
 import { NotRecognizedIcon } from "@/components/icons/NotRecognizedIcon";
 import { RecognizedIcon } from "@/components/icons/RecognizedIcon";
 import { MarketIconPair } from "@/components/MarketIconPair";
 import { useAptos } from "@/contexts/AptosContext";
-import { type ApiMarket, type ApiStats } from "@/types/api";
+import { type ApiMarket } from "@/types/api";
 import { formatNumber, plusMinus } from "@/utils/formatter";
 import { TypeTag } from "@/utils/TypeTag";
 
 import { useAllMarketStats } from ".";
-const columnHelper = createColumnHelper<ApiMarket>();
 
-const TABLE_SPACING = {
-  margin: "-mx-6 -mb-6",
-  paddingLeft: "pl-6",
-  paddingRight: "pr-6",
+const DEFAULT_TOKEN_ICON = "/tokenImages/default.png";
+const colWidths = [260, undefined, undefined, 120, 130] as const;
+
+type MarketWithStats = {
+  marketId: number;
+  baseSymbol?: string;
+  quoteSymbol: string;
+  baseAssetIcon: string;
+  quoteAssetIcon: string;
+  name: string;
+  price?: number;
```

Comment:
Hmmmm I can look into this, but having them as base level fields might make it easier to take advantage of TanStack table features. There might not be too much advantage of being able to batch these since you still deal with them individually in the `cell` function for each column